### PR TITLE
Add-to-app iOS local network permission

### PR DESCRIPTION
**Pull request by jmagman**
Add instructions for allowing iOS 14 local networking permissions, as promised at the end of https://flutter.dev/docs/development/ios-14#debugging-flutter.

Details in the `samples` PR where I followed these instructions: https://github.com/flutter/samples/pull/554

Fixes https://github.com/flutter/flutter/issues/63893.

### CODE REVIEW

File: src/docs/development/add-to-app/ios/project-setup.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ description: Learn how to integrate a Flutter module into your existing iOS proj
 Flutter can be incrementally added into your existing iOS
 application as embedded frameworks.
 
+For examples, see the [add_to_app code samples][].
```
Comment (xster) [inline src/docs/development/add-to-app/ios/project-setup.md:10]: see the iOS directories in the [add_to_app code samples][]?

File: src/docs/development/add-to-app/ios/project-setup.md [hunk 2]
```diff
@@ -344,11 +346,80 @@ FlutterPluginRegistrant.framework,
 and any plugin frameworks into your existing application
 as described in Option B.
 
+## Local Network Privacy Permissions
+On iOS 14 and higher, enable the Dart multicast DNS
+service in the Debug version of your app
+to allow Flutter debugging functionalities such as hot-reload and DevTools.
```
Comment (xster) [inline src/docs/development/add-to-app/ios/project-setup.md:352]: maybe add debugging functionalities such as hot reload and DevTools via `flutter attach`. See more on debugging in https://flutter.dev/docs/development/add-to-app/debugging.

**Review by xster — COMMENTED** [review]
LGTM. Thanks!

File: src/docs/development/add-to-app/ios/project-setup.md (hunk 1 above)
Comment (jmagman) [reply src/docs/development/add-to-app/ios/project-setup.md]: 👍

File: src/docs/development/add-to-app/ios/project-setup.md (hunk 2 above)
Comment (jmagman) [reply src/docs/development/add-to-app/ios/project-setup.md]: Changed it to 
```
to add [debugging functionalities such as hot-reload and DevTools][] via `flutter attach`.
```
with that link.

**Review by xster — APPROVED** [review]

**Comment by xster** [comment]
Since we added a new header, could we change the link in https://flutter.dev/docs/development/ios-14#debugging-flutter too?

**Comment by jmagman** [comment]
> Since we added a new header, could we change the link in https://flutter.dev/docs/development/ios-14#debugging-flutter too?

Good idea, done.

**Review by sfshaza2 — APPROVED** [review]
LGTM

**Pull request merged**